redash/handlers/data_sources.py

```diff
@@ -84,13 +84,15 @@ def get(self):
         else:
             data_sources = models.DataSource.all(self.current_org, group_ids=self.current_user.group_ids)
 
+        all_fields = request.args.get('extended', None) is not None
+
         response = {}
         for ds in data_sources:
             if ds.id in response:
                 continue
 
             try:
-                d = ds.to_dict()
+                d = ds.to_dict(all=all_fields)
```

Comment:
This is to get the groups information?

Reply:
Yes. I decided to add a parameter so it doesn't break any existing code (if any).

Reply:
You should be careful when doing such changes -- in this case, you're now exposing all data source settings to all users. Previously we were showing this only to admins. While in theory all secrets are masked, in practice for some data sources not all secrets are masked.

You can add another `include_groups` option, and use it here to add just the groups.

But why do you need this in the first place?

Reply:
@arikfr 

> But why do you need this in the first place?

I need it in "Add data sources" dialog - to mark items that are already added (similar as in "Add members". I'll re-implement it - will add just a list of groups as array.

Reply:
But you already have the list of data sources in this group on the page that opens this modal, can't you pass this information to it? 

While you can update the API to include groups, it's just extra queries that we don't really need.

Reply:
@arikfr How about this? https://github.com/getredash/redash/pull/3411/commits/92c57e9058894033af177099306e4c4d4857740e

Reply:
@arikfr Reverted all changes to backend, you're right - I totally forgot that I already have all items :man_facepalming: 

https://github.com/getredash/redash/blob/fd0659d991f05adb27a2253b8a654cabfdde3d2e/client/app/pages/groups/GroupDataSources.jsx#L126-L128

Reply:
Implemented it in the same way for Add members dialog too